### PR TITLE
Refator/teacher dashboard

### PR DESCRIPTION
**Pull request by VictorDronov**
# Refactor and feature addition for teacher display when no class/rumbles and students

### CODE REVIEW

**Review by sjsussman — APPROVED** [review]

File: src/styles/components/pages/teacherDashboard/teacherDashboardStudentList.scss [hunk 1]
```diff
@@ -6,6 +6,33 @@
       @include bp(tablet) {
         @include table(0, 35%, 35%, 30%);
       }
+      // ***
+      div {
```
Comment (bramirez96) [inline src/styles/components/pages/teacherDashboard/teacherDashboardStudentList.scss:10]: Let's not use `div` as a selector, it's not specific enough. Most of the time there should be a class name here

File: src/styles/components/pages/teacherDashboard/teacherDashboardSectionList.scss [hunk 2]
```diff
@@ -33,6 +34,35 @@
         border: 1px solid $c-grey-20;
       }
     }
+    .no-sections {
+      & > div {
+        display: flex;
+        flex-flow: column wrap;
+        overflow-x: scroll;
+        overflow-y: hidden;
+        .speach {
```
Comment (bramirez96) [inline src/styles/components/pages/teacherDashboard/teacherDashboardSectionList.scss:43]: speech <--

File: src/styles/components/pages/teacherDashboard/teacherViewRumblePage/teacherRumbleStudentList.scss [hunk 3]
```diff
@@ -6,6 +6,34 @@
       @include bp(tablet) {
         @include table(0, 35%, 35%, 30%);
       }
+      // *** This probably can be but in its own file along with the teacherDashboardStudentList style for the same things
```
Comment (bramirez96) [inline src/styles/components/pages/teacherDashboard/teacherViewRumblePage/teacherRumbleStudentList.scss:9]: Let's get a class name so it's easier to see what this is actually styling

File: src/assets/img/Speach.svg [hunk 4]
```diff
@@ -0,0 +1,5 @@
+<svg width="230" height="132" viewBox="0 0 230 132" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (bramirez96) [inline src/assets/img/Speach.svg:1]: change the name to `speech` (lowercase)

File: src/assets/img/no-students.svg [hunk 5]
```diff
@@ -0,0 +1,53 @@
+<svg width="400" height="300" viewBox="0 0 400 300" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (bramirez96) [inline src/assets/img/no-students.svg:1]: let's make non-typescript file names all snake case? `no_students`

File: src/assets/img/norumbles.svg [hunk 6]
```diff
@@ -0,0 +1,35 @@
+<svg width="286" height="233" viewBox="0 0 286 233" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (bramirez96) [inline src/assets/img/norumbles.svg:1]: no_rumbles

File: src/components/pages/TeacherDashboard/TeacherDashboardRumbleList/TeacherDashboardRumbleList.tsx [hunk 7]
```diff
@@ -1,27 +1,49 @@
...
+import React, { useEffect, useState } from 'react';
 import { Sections } from '../../../../api';
+import emptyMail from '../../../../assets/img/norumbles.svg';
 import TeacherDashboardRumbleCard from './TeacherDashboardRumbleCard';
 
 const TeacherDashboardRumbleList = ({
   sections,
   visible = true,
 }: ITeacherDashboardRumbleListProps): React.ReactElement => {
+  const [rumbleLength, setRumbleLength] = useState<number>();
```
Comment (bramirez96) [inline src/components/pages/TeacherDashboard/TeacherDashboardRumbleList/TeacherDashboardRumbleList.tsx:10]: let's brainstorm a better way to handle this

File: src/components/pages/TeacherDashboard/TeacherDashboardSectionList/TeacherDashboardSectionList.tsx [hunk 8]
```diff
@@ -13,28 +16,48 @@ const RenderTeacherDashboardSectionList = ({
   const [newSectionOpen, setNewSectionOpen] = useState(false);
   const openSectionModal = () => setNewSectionOpen(true);
   const openInviteModal = () => setInviteModalOpen(true);
+  const sectionList = useRecoilValue(list);
```
Comment (bramirez96) [inline src/components/pages/TeacherDashboard/TeacherDashboardSectionList/TeacherDashboardSectionList.tsx:19]: sections (props) vs sectionList (recoil) we need to make sure we're using the correct list

File: src/components/pages/TeacherDashboard/TeacherViewRumble/RumbleStudentList/RenderRumbleStudentList.tsx [hunk 9]
```diff
@@ -1,31 +1,53 @@
...
-          <Table.Body>
-            {studentList.map((student) => (
-              <RumbleStudentCard
-                student={student}
-                section={section}
-                key={student.id}
-              />
-            ))}
-          </Table.Body>
+          <InviteToSection />
```
Comment (bramirez96) [inline src/components/pages/TeacherDashboard/TeacherViewRumble/RumbleStudentList/RenderRumbleStudentList.tsx:21]: I think the modal is already being displayed somewhere else and we don't need to put in in again, that's what the recoil state is for

**Review by bramirez96 — CHANGES_REQUESTED** [review]
Generally looks good, a few file name changes and some stuff that we should definitely go over together and make sure it's working correctly and efficiently

**Pull request closed without merging**